mathjax3-ts/input/tex.ts

```diff
@@ -22,58 +22,323 @@
+  /**
+   * Name of input jax.
+   * @type {string}
+   */
+  public static NAME: string = 'TeX';
+
+  /**
+   * Default options for the jax.
+   * @type {OptionList}
+   */
+  public static OPTIONS: OptionList = {
+    ...AbstractInputJax.OPTIONS,
+    FindTeX: null,
+    packages: ['base'],
+    settings: {
+      //  This specifies the side on which \tag{} macros will place the tags.
+      //  Set to 'left' to place on the left-hand side.
+      TagSide: 'right',
+      //  This is the amound of indentation (from right or left) for the tags.
+      TagIndent: '0.8em',
+      //  This is the width to use for the multline environment
+      MultLineWidth: '85%',
+      // make element ID's use \label name rather than equation number
+      // MJ puts in an equation prefix: mjx-eqn
+      // When true it uses the label name XXX as mjx-eqn-XXX
+      // If false it uses the actual number N that is displayed: mjx-eqn-N
+      useLabelIds: true,
+      refUpdate: false
+    },
+    // Tagging style, used to be autonumber in v2.
+    tags: 'none'
+  };
+
+  /**
+   * The FindTeX instance used for locating TeX in strings
+   */
+  protected findTeX: FindTeX<N, T, D>;
+
+  private configuration: Configuration;
+  private parseOptions: ParseOptions;
+  private latex: string;
+  private mathNode: MmlNode;
+  
+  /**
+   * Wraps an error into a node for output.
+   * @param {TeXError} err The TexError.
+   * @return {Node} The merror node.
+   */
+  private formatError(err: TexError): MmlNode {
+    let message = err.message.replace(/\n.*/, '');
+    return this.parseOptions.createNode('error', message);
+  };
+
+
+  /**
+   * Visitor to set stretchy attributes to false on <mo> elements, if they are
+   * not used as delimiters. Also wraps non-stretchy infix delimiters into a
+   * TeXAtom.
+   * @param {MmlNode} node The node to rewrite.
+   * @param {ParseOptions} options The parse options.
+   */
+  private static cleanStretchy(node: MmlNode, options: ParseOptions) {
+    node.walkTree((mo: MmlNode, d: any) => {
+      if (NodeUtil.getProperty(mo, 'fixStretchy')) {
+        let symbol = NodeUtil.getForm(mo);
+        if (symbol && symbol[3] && symbol[3]['stretchy']) {
+          NodeUtil.setAttribute(mo, 'stretchy', false);
+        }
+        const parent = mo.parent;
+        if (!NodeUtil.getTexClass(mo) && (!symbol || !symbol[2])) {
+          const texAtom = options.nodeFactory.create('node', 'TeXAtom', [mo], {});
+          texAtom.parent = parent;
+          parent.replaceChild(texAtom, mo);
+        }
+        NodeUtil.removeProperties(mo, 'fixStretchy');
+      }
+    }, {});
+  }
+
+
+  /**
+   * Visitor that removes superfluous attributes from nodes. I.e., if a node has
+   * an attribute, which is also an inherited attribute it will be removed. This
+   * is necessary as attributes are set bottom up in the parser.
+   * @param {MmlNode} mml The node to clean.
+   * @param {ParseOptions} options The parse options.
```

Comment:
This was because all the `(post/pre)processors` had to have the same type. 
Both are now moved into `pre/postFilters` and use the dedicated caller of the `InputJax` class.
Filters are also now in a new `FilterUtil` namespace which make this file much cleaner.